### ISSUE TITLE
[SECURITY] Uploading HTML files could lead to user redirection

### ISSUE THREAD

**Issue by bryceatmandiant** [galaxyproject/galaxy#14631]
**Describe the bug**
We were able to upload an HTML file that renders HTML content, specifically an HTML form. A submit button can be used in this form to forward a user to an arbitrary URL. 

**Galaxy Version and/or server at which you observed the bug**
Galaxy Version: 22.09.dev0 (latest)

**Steps to reproduce the behavior:**

1. Create an HTML file with this content:
`<html><form action="http://arbitrarydomainatgalaxyproject.com"><input type="submit" value="Redirect to Arbitrary Domain"></form></html>`

2. Upload the file through the "Upload Data" button 
![image](https://user-images.githubusercontent.com/100627745/190502353-66694011-38f9-4b00-8e18-9228824bc91e.png)

3. Press the display button on the uploaded file (the eye icon)
![image](https://user-images.githubusercontent.com/100627745/190502455-e82697c9-a4e2-438f-8b36-fc812e1d8e01.png)

4. Click the Redirect the button to submit the form 
![image](https://user-images.githubusercontent.com/100627745/190502517-f7b1b660-655b-4bf7-a64c-4079525a2486.png)

**Expected behavior**
The best solution is that no HTML files should be uploaded. However, I understand cases where genome data may be organized in an HTML file. I recommend that form elements be disallowed in uploaded HTML files. Another solution is to create a page can be used for whenever a redirection happens from the Galaxy Project application. This could look something like Google's redirect page on their outstanding open redirect vulnerability [at this link](https://meet.google.com/linkredirect?dest=https://accounts-google.phishy.info/).
![image](https://user-images.githubusercontent.com/100627745/190503786-f4d2f3f6-305f-427b-afcc-3a3689bc8af5.png)

**Additional context**
I was unsuccessful in escalating this to an XSS because it was stripping any JavaScript functions/content, so at first glance, this appears good at stopping an XSS attack.

Also on a side note, I noticed [this file has a function](https://github.com/galaxyproject/galaxy/blob/9143dd7ca46d150ebfb26febbe187979f682da51/lib/galaxy/datatypes/dataproviders/external.py) that appears very vulnerable to [Command Injection](https://owasp.org/www-community/attacks/Command_Injection). I would have to spend a longer time assessing this project and try to see how I can reach this source, but a second review on this function could help prevent a very serious security vulnerability.
![image](https://user-images.githubusercontent.com/100627745/190505548-6678050a-e219-436d-8b29-4c032e790454.png)

**Comment by bryceatmandiant** [comment]
I did not think this even raised to a Medium risk rating on y'all's scale in the SECURITY.MD, so I wrote this out as an issue instead.

**Comment by hexylena** [comment]
You're right, this _seems_ relatively low risk. Yes we support HTML uploads as long as they are sanitized, no unsanitized uploads can be allowed. For this specific case the major risk boils down to [`noopener`](https://developer.mozilla.org/en-US/docs/Web/HTML/Link_types/noopener), but either way it requires user interaction to do a redirection which, yeah. I'd support the google style interception, if we can imitate that somehow.

(I cannot speak to the subprocess issue, but I don't recognise that class as something being heavily used)

**Comment by bryceatmandiant** [comment]
Yea you're right, this does require user interaction, and therefore, it's mostly going to be used in phishing attacks.
